src/BaGetter.Database.Sqlite/SqliteContext.cs

```diff
@@ -45,5 +52,75 @@ protected override void OnModelCreating(ModelBuilder builder)
                 .Property(f => f.Moniker)
                 .HasColumnType("TEXT COLLATE NOCASE");
         }
+
+        public override async Task RunMigrationsAsync(CancellationToken cancellationToken)
+        {
+            if (Database.GetDbConnection() is SqliteConnection connection)
+            {
+                /* Create the folder of the Sqlite blob if it does not exist. */
+                CreateSqliteDataSourceDirectory(connection);
+            }
+
+            await base.RunMigrationsAsync(cancellationToken);
+        }
+
+        /// <summary>
+        /// Creates directories specified in the Database::ConnectionString config for the Sqlite database file.
+        /// Respects the difference between Windows and other OS root or relative paths.
+        /// </summary>
+        /// <param name="connection">Instance of the <see cref="SqliteConnection"/>.</param>
+        private static void CreateSqliteDataSourceDirectory(SqliteConnection connection)
+        {
+            var configIsLinuxRootPath = connection.DataSource.StartsWith('/');
+            var configDataSource = connection.DataSource;
+
+            string? pathToCreate = null;
+
+            if (OperatingSystem.IsWindows())
+            {
+                var windowsDriveLetter = Path.GetPathRoot(Directory.GetCurrentDirectory())!;
+
+                // root path
+                if (configIsLinuxRootPath)
+                {
+                    var directories = Path.GetDirectoryName(configDataSource);
+                    if (directories?.Length > 0) pathToCreate = Path.Combine(windowsDriveLetter, directories);
+                }
+                else if (WindowsDriveLetterRegex.IsMatch(configDataSource))
+                {
+                    pathToCreate = Path.GetDirectoryName(configDataSource);
+                }
+                else
+                {
+                    pathToCreate = CreateFullDataSourcePath();
+                }
+            }
+            else
+            {
+                pathToCreate = configIsLinuxRootPath
+                    // root path
+                    ? Path.GetDirectoryName(configDataSource)
+                    // relative path
+                    : CreateFullDataSourcePath();
+            }
+
+            if (pathToCreate is not null)
+            {
+                Directory.CreateDirectory(pathToCreate);
+            }
+
+            return;
+
+            string CreateFullDataSourcePath()
+            {
+                var exeDir = Directory.GetCurrentDirectory();
+                var relativePath = Path.GetDirectoryName(configDataSource);
+                var relativeDirectoriesArray = relativePath?.Split(Path.DirectorySeparatorChar) ?? [];
+                var relativeDirectoriesCombined = Path.Combine(relativeDirectoriesArray);
+                var fullPath = Path.Combine(exeDir, Path.Combine(relativeDirectoriesCombined));
```

Comment:
Is the inner `Path.Combine` necessary? What does it even do on the output of a previous `Path.Combine`?

Reply:
I'll check if this makes a difference